### PR TITLE
[test] check repository when using git auto-update

### PR DESCRIPTION
**Pull request by pvnr0082t**
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required):
 * Official website (optional, not the repository):
 * NPM package url (optional):
 * GitHub / Bitbucket popularity (required):
   - Count of stars:
   - Count of watchers:
   - Count of forks:
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

add one check point to check repo info when using git auto-update, cc @PeterDaveHello

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
@pvnr0082t since this change will detect some problems and then break the build, please also fix the libs with missing repo info, you can simply use its auto-update config to fill the repo info.

Maybe libs with npm auto-update should also have repo info, but I'm not sure how many libs that has npm auto-update config doesn't have repo info, should make sure this first, then we can add both npm and git autoupdate check in this PR.

**Comment by pvnr0082t** [comment]
@PeterDaveHello There are 14 broken show on CI. So I only need to fix these libs first or I still need to check all libs with git auto-update?

**Comment by PeterDaveHello** [comment]
@pvnr0082t You can fix it before or after the new test, didn't the build check all the libs for you?

**Comment by PeterDaveHello** [comment]
#7938 merged, let's update this branch and see if it can pass the test.

**Comment by pvnr0082t** [comment]
@PeterDaveHello The test passed. :+1:

File: test/valid-packages-test.js [hunk 1]
```diff
@@ -288,6 +288,11 @@ packages.map(function (pkg) {
         assert.ok(JSON.stringify(json) === JSON.stringify(json_fix) ,
             pkg_name(pkg) + ": we don't need bin, jshintConfig, eslintConfig, styles, install, typescript, browserify, browser, jam, jest, scripts, devDependencies, main, peerDependencies, contributors, bugs, gitHEAD, issues, files, ignore, engines, engine, directories and maintainers fields in package.json");
     }
+    package_vows[pname + ": There must be repository information when using git auto-update config"] = function(pkg) {
+        var json = parse(pkg, true, true);
+        assert.ok(json.autoupdate === undefined || json.repository != undefined,
+        pkg_name(pkg) + ": Need to add repository information in package.json");
```
Comment (PeterDaveHello) [inline test/valid-packages-test.js]: Wrong indent, please add 4 spaces here.

**Comment by pvnr0082t** [comment]
@PeterDaveHello Indent updated. Please review again, thank you.

**Pull request merged**